File: ChordVoicingsCreator/src/com/campoy/chord/voicing/creator/model/CircularLinkedListNode.java
package com.campoy.chord.voicing.creator.model;

public class CircularLinkedListNode <T> {
    
    T value;
    CircularLinkedListNode<T> nextNode;
 
    public CircularLinkedListNode(T value) {
        this.value = value;
    }
    
    @Override
    public String toString() {
        return "Node(" + value.toString() + ")";
    }
    
    @Override
    public boolean equals(Object obj) {
        
        if(obj == null || !(this.getClass().isAssignableFrom(obj.getClass())))
        {               
            return false;
        }
        
        CircularLinkedListNode<T> other = (CircularLinkedListNode<T>) obj;
        
        if(this.value == null 
                && other.value == null)
        {
            return true;
        }
        else if (this.value == null)
        {
            return false;
        }

        return value.equals(other.value);
    }
}

File: PancakeMusicProjectsTS/src/model/musictheory/CircularLinkedListNode.ts
import { EqualsTrait } from "../general/EqualsTrait";

export class CircularLinkedListNode<T extends EqualsTrait> implements EqualsTrait{
    value: T;
    nextNode: CircularLinkedListNode<T> | null;
  
    constructor(value: T) {
      this.value = value;
      this.nextNode = null;
    }
  
    toString(): string {
      return `Node(${this.value})`;
    }
  
    equals(obj: any): boolean {
      if (!(obj instanceof CircularLinkedListNode)) {
        return false;
      }
    
      if (this.value === null && obj.value === null) {
        return true;
      } else if (this.value === null) {
        return false;
      }
  
      return this.value.equals(obj.value);
    }
  }